backends/gstreamer/lib.rs

```diff
@@ -152,3 +188,246 @@ impl<'a> Drop for MutexGuard<'a> {
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_DECODER |
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_MEDIA_AUDIO,
+            gst::Rank::Marginal,
+        );
+        let audio_parser_factories = gst::ElementFactory::list_get_elements(
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_PARSER |
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_MEDIA_AUDIO,
+            gst::Rank::None,
+        );
+        let video_decoder_factories = gst::ElementFactory::list_get_elements(
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_DECODER |
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_MEDIA_VIDEO,
+            gst::Rank::Marginal,
+        );
+        let video_parser_factories = gst::ElementFactory::list_get_elements(
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_PARSER |
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_MEDIA_VIDEO,
+            gst::Rank::Marginal,
+        );
+        let demux_factories = gst::ElementFactory::list_get_elements(
+            gst_ffi::GST_ELEMENT_FACTORY_TYPE_DEMUXER,
+            gst::Rank::Marginal,
+        );
+
+        if has_element_for_media_type(&audio_decoder_factories, "audio/mpeg, mpegversion=(int)4") {
+            self.supported_mime_types.insert("audio/aac");
+            self.supported_mime_types.insert("audio/mp4");
+            self.supported_mime_types.insert("audio/x-m4a");
+            self.supported_codecs.insert("mpeg");
+            self.supported_codecs.insert("mp4a*");
+        }
+
+        let is_opus_supported = has_element_for_media_type(&audio_decoder_factories, "audio/x-opus");
+        if is_opus_supported && has_element_for_media_type(&audio_parser_factories, "audio/x-opus") {
+            self.supported_mime_types.insert("audio/opus");
+            self.supported_codecs.insert("opus");
+            self.supported_codecs.insert("x-opus");
+        }
+
+        let is_vorbis_supported = has_element_for_media_type(&audio_decoder_factories, "audio/x-vorbis");
+        if is_vorbis_supported && has_element_for_media_type(&audio_parser_factories, "audio/x-vorbis") {
+            self.supported_codecs.insert("vorbis");
+            self.supported_codecs.insert("x-vorbis");
+        }
+
+        if has_element_for_media_type(&demux_factories, "video/x-matroska") {
+            let is_vp8_decoder_available = has_element_for_media_type(&video_decoder_factories, "video/x-vp8");
+            let is_vp9_decoder_available = has_element_for_media_type(&video_decoder_factories, "video/x-vp9");
+
+            if is_vp8_decoder_available || is_vp9_decoder_available {
+                self.supported_mime_types.insert("video/webm");
+            }
+
+            if is_vp8_decoder_available {
+                self.supported_codecs.insert("vp8");
+                self.supported_codecs.insert("x-vp8");
+            }
+
+            if is_vp9_decoder_available {
+                self.supported_codecs.insert("vp9");
+                self.supported_codecs.insert("x-vp9");
```

Comment:
And vp9.0 here :)
I've fixed it in WebKit: https://bugs.webkit.org/show_bug.cgi?id=196615